File: 0001-two-sum/0001-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {

    const hash = new Map();

    for(let i=0 ; i <nums.length ; i++){
        let rest = target - nums[i];

         if (hash.has(rest)) {
            return [hash.get(rest), i];
        } else {
            hash.set(nums[i], i);
        }
    }

    throw new Error("No solution")
    
};

File: 0001-two-sum/0001-two-sum.java
import java.util.HashMap;
import java.util.Map;

class Solution {
    public int[] twoSum(int[] nums, int target) {

        HashMap<Integer,Integer> hash = new HashMap<>();

        for(int i=0 ; i < nums.length ; i++){
            int diff = target - nums[i];

            if (hash.containsKey(diff)) {
                return new int[] {hash.get(diff), i};
            }

            hash.put(nums[i], i);
        }

        throw new IllegalArgumentException("No solution");


        
    }
}
